pkg/jsontabwriter/jsontabwriter.go

```diff
@@ -0,0 +1,204 @@
+		return generateTextOutputFromJSON(rootPath, obj, jsonPaths, cols)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func GenerateOutputPreconverted(obj interface{}, convertedObj []map[string]interface{}, cols []string) (string, error) {
+	if viper.IsSet(constants.ArgQuiet) {
+		return "", nil
+	}
+
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		return generateJSONOutput(obj)
+	}
+
+	if outputFormat == "text" {
+		return generateTextOutputFromTable(convertedObj, cols)
+	}
+
+	return "", fmt.Errorf(outputFormatErr, outputFormat)
+}
+
+func GenerateVerboseOutput(format string, a ...interface{}) string {
+	if viper.IsSet(constants.ArgQuiet) {
+		return ""
+	}
+
+	if !viper.IsSet(constants.ArgVerbose) {
+		return ""
+	}
+
+	msg := fmt.Sprintf("[INFO] "+format, a...)
+
+	return GenerateRawOutput(msg)
+}
+
+func GenerateLogOutput(format string, a ...interface{}) string {
+	msg := fmt.Sprintf(format, a...)
+
+	return GenerateRawOutput(msg)
+}
+
+func GenerateRawOutput(a interface{}) string {
+	if viper.IsSet(constants.ArgQuiet) {
+		return ""
+	}
+
+	outputFormat := viper.GetString(constants.ArgOutput)
+
+	if outputFormat == "json" {
+		out, err := generateJSONOutput(a)
+		if err != nil {
+			return ""
+		}
+
+		return out
+	}
+
+	if outputFormat == "text" {
+		return fmt.Sprintf("%v\n", a)
+	}
+
+	return ""
+}
+
+func generateJSONOutput(obj interface{}) (string, error) {
+	out, err := json.MarshalIndent(obj, "", "\t")
+	if err != nil {
+		return "", err
+	}
+
+	return string(out) + "\n", nil
+}
+
+func generateTextOutputFromJSON(rootPath string, obj interface{}, jsonPaths map[string]string, cols []string) (string, error) {
+	table, err := json2table.ConvertJSONToTable(rootPath, jsonPaths, obj)
+	if err != nil {
+		return "", fmt.Errorf("failed converting object to table using %+v: %w", jsonPaths, err)
+	}
+
+	return writeTableToText(table, cols), nil
+}
+
+func generateTextOutputFromTable(table []map[string]interface{}, cols []string) (string, error) {
+	return writeTableToText(table, cols), nil
+}
+
+func writeTableToText(table []map[string]interface{}, cols []string) string {
+	var buff = new(bytes.Buffer)
+	var w = new(tabwriter.Writer)
+
+	w.Init(buff, 5, 0, 3, ' ', tabwriter.StripEscape)
+
+	updatedCols := eliminateEmptyCols(cols, table)
+
+	if !viper.IsSet(constants.ArgNoHeaders) {
+		fmt.Fprintln(w, strings.Join(updatedCols, "\t"))
+	}
+
+	for _, t := range table {
+		format, values := convertTableToText(updatedCols, t)
+		fmt.Fprintf(w, format+"\n", values...)
+	}
+
+	w.Flush()
+
+	return buff.String()
+}
+
+func convertTableToText(cols []string, table map[string]interface{}) (format string, values []interface{}) {
+	formats := make([]string, 0)
+	values = make([]interface{}, 0)
+
+	for _, col := range cols {
+		field := table[col]
+		formats = append(formats, "%v")
+
+		if field == nil {
+			values = append(values, "")
+
+			continue
+		}
+
+		switch field.(type) {
```

Comment:
(nitpicking) 

You can reduce the repetition of the types here by doing `switch fieldType := field.(type) {`

This would mean you would have less type casts. But the current version is fine too.